test/test_private_offer_stage.js

```diff
@@ -0,0 +1,76 @@
+var Token = artifacts.require("MFC_Token");
+var Controller = artifacts.require("ICO_controller");
+var WhitelistedCrowdsale = artifacts.require("WhitelistedCrowdsale");
+var BigNumber = require('bignumber.js');
+var wait = require('./utils').wait;
+var Web3 = require('web3');
+var web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
+
+contract('Private Offer', async function (accounts) {
```

Comment:
This test should concentrate on `buyTokens` testing, private offer (or other stage) is less important here, since `buyTokens` is in the `WhitelistedCrowdsale` contract which is the same for all stages.

So I think it should be: file `test_ico_crowdsale`, then `contract(Buy tokens', async function (accounts) {...` and `it("Allows to buy tokens from the private offer stage controller")`.